CMakeLists.txt

```diff
@@ -9,12 +9,64 @@ set(SYSREPO_MICRO_VERSION 10)
 set(SYSREPO_VERSION ${SYSREPO_MAJOR_VERSION}.${SYSREPO_MINOR_VERSION}.${SYSREPO_MICRO_VERSION})
 
 # setup bindings
-SET(GEN_LANGUAGE_BINDINGS 1 CACHE BOOL "Enable language bindings generation.")
-SET(GEN_PYTHON_BINDINGS 1 CACHE BOOL "Enable python bindings.")
-SET(GEN_LUA_BINDINGS 1 CACHE BOOL "Enable Lua bindings.")
-SET(GEN_CPP_BINDINGS 1 CACHE BOOL "Enable C++ bindings.")
-SET(GEN_PYTHON_VERSION "2")
-SET(GEN_LUA_VERSION "5.1")
+set(GEN_LANGUAGE_BINDINGS 1 CACHE BOOL "Enable language bindings generation.")
+set(GEN_PYTHON_BINDINGS 1 CACHE BOOL "Enable python bindings.")
+set(GEN_LUA_BINDINGS 1 CACHE BOOL "Enable Lua bindings.")
+set(GEN_CPP_BINDINGS 1 CACHE BOOL "Enable C++ bindings.")
+set(GEN_PYTHON_VERSION "3")
+set(GEN_LUA_VERSION "5.2")
+
+# find SWIG package
+if(GEN_LANGUAGE_BINDINGS)
+    find_package(SWIG)
+    if(NOT SWIG_FOUND)
+        message(WARNING "SWIG library not found")
+    else()
+        include(${SWIG_USE_FILE})
+    endif()
+endif()
+
+# find Lua package
+if(GEN_LUA_BINDINGS AND SWIG_FOUND)
```

Comment:
Mislav, please, just to make the main CMakeLists little bit better readable (it becomes too large): could we move all this new stuff (or better all swig-related stuff) to the `swig/CMakeLists.txt` file? The only thing that I would leave here would be variables - GEN_LANGUAGE_BINDINGS, GEN_PYTHON_BINDINGS, etc.